fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Channel.java

```diff
@@ -0,0 +1,132 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class represent a channel used to subscribe to a certain part of the messages. More detailed
+ * information are given in the protocol specification of the application.
+ *
+ * <p>It is implemented in a functional manner to avoid any unintended modification
+ */
+public class Channel {
+
+  private static final String ROOT_CHANNEL = "/root";
+  private static final String DELIMITER = "/";
+  // Predicate matching a String with the protocol regex of a channel
+  private static final Predicate<String> CHANNEL_VALIDATOR =
+      Pattern.compile("^/root(/[^/]+)*$").asPredicate();
+
+  /** Root channel, every client is subscribed to it. */
+  public static final Channel ROOT = new Channel();
+
+  public static Channel newChannel(String value) {
+    // Avoid creating duplicate instances
+    if (ROOT_CHANNEL.equals(value)) return ROOT;
+
+    // Remove the "/root/" part of the value as it is not relevant and does not need to be stored
+    String relevantPart = value.substring(ROOT_CHANNEL.length() + 1);
+    return new Channel(relevantPart.split(DELIMITER));
+  }
+
+  // ================ Object itself =================
+
+  // This list contains the different parts of the channel except the root as every channel has a
+  // root part
+  // So for the channel /root/lao_id/election_id, the elements will be [lao_id, election_id]
+  @NonNull private final List<String> elements;
+
+  private Channel(String... elements) {
+    this.elements = Arrays.asList(elements);
```

Comment:
since we're using a classic URI path format, I would suggest calling them segments (the classical term), rather than elements.

Reply:
Yes, good idea